### PR TITLE
Fixed System.Runtime.Extensions uap/uapaot/netfx tests (combined with Viktor's PR)

### PR DESCRIPTION
**Pull request by krwq**
Contributes to: https://github.com/dotnet/corefx/issues/18718
This is resolving conflicts with mostly overlapping changes with @danmosemsft's PR: https://github.com/dotnet/corefx/pull/21412
I've left original PR for comparison: https://github.com/dotnet/corefx/pull/19866

We might want to revert changes to src/System.Runtime.Extensions/tests/System/AppDomainTests.cs

cc: @ViktorHofer 

Testing in progress so may fail CI

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Did you take that one in consideration? https://github.com/dotnet/corefx/pull/19866#discussion_r118706701

**Comment by krwq** [comment]
@ViktorHofer, thanks - didn't notice that, will fix as soon as my local build is finished

**Comment by krwq** [comment]
@ViktorHofer seems like there is nothing to change there as the test case is already enabled. Seems like this conflict must have auto-resolved correctly

**Comment by krwq** [comment]
@dotnet-bot test NETFX x86 Release Build please

**Comment by krwq** [comment]
@danmosemsft S.R.Extesnions pass UAP locally, I haven't run all other tests

**Comment by danmosemsft** [comment]
OK LGTM

**Comment by krwq** [comment]
CI is stuck? this is taking a long

**Comment by danmosemsft** [comment]
@dotnet-bot test NETFX x86 Release Build please (S.R.E tests hung.. perhaps due to this change)

**Comment by danmosemsft** [comment]
@krwq please run netfx tests locally.

**Comment by ViktorHofer** [comment]
@dotnet-bot test NETFX x86 Release Build

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 1]
```diff
@@ -2,19 +2,15 @@
...
 using System.Reflection;
 using System.Resources;
 using System.Runtime.ExceptionServices;
 using Xunit;
-using Xunit.NetCore.Extensions;
 
 namespace System.Tests
 {
-    // No appdomain in UWP or CoreRT
-    [SkipOnTargetFramework(TargetFrameworkMonikers.UapAot | TargetFrameworkMonikers.NetFramework, "dotnet/corefx #18718")]
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: We might need to investigate why the class was skipped before on netfx.

Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: It seems one/multiple methods are blocking.

**Comment by krwq** [comment]
Ok, I have identified what is hanging and also found few issues on netfx I have already fixed. I'll update this PR once I figure out which cases do we want to test on netfx and which we don't

**Comment by krwq** [comment]
Ok, I fixed netfx tests where I could, some items I left disabled. This PR has significantly changed so please review

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 2]
```diff
@@ -167,97 +188,124 @@ public FirstChanceTestException(string message) : base(message)
...
+                AssemblyName assemblyName = assembly.GetName();
+                assemblyName.CodeBase = null;
+                Assert.Equal(105, AppDomain.CurrentDomain.ExecuteAssemblyByName(assemblyName, new string[3] { "50", "25", "25" }));
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         [ActiveIssue("https://github.com/dotnet/corefx/issues/18718", TargetFrameworkMonikers.Uap)] // Need to copy files out of execution directory
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Access issues when copying assemblies")]
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: What kind of access issues?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 3]
```diff
@@ -268,28 +316,43 @@ public void IsCompatibilitySwitchSet()
         [Fact]
         public void IsDefaultAppDomain()
         {
-            Assert.True(AppDomain.CurrentDomain.IsDefaultAppDomain());
+            RemoteInvoke(() =>
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Is a RemoteExecutor required here because the xunit appdomain where the tests are running in isn't the default one?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 4]
```diff
@@ -41,45 +37,54 @@ public void RelativeSearchPath_Is_Null()
...
         {
-            System.IO.File.Delete("success.txt");
-            RemoteInvokeOptions options = new RemoteInvokeOptions();
-            options.CheckExitCode = false;
-            RemoteInvoke(() =>
-            {
-                AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(MyHandler);
-                throw new Exception("****This Unhandled Exception is Expected****");
+            RemoteInvoke(() => {
+                System.IO.File.Delete("success.txt");
```
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Doesn't this fail in app container? It should be in %temp%, eg by using System.IO.TempFile from src\Common\tests\System\IO\TempFile.cs

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 5]
```diff
@@ -268,28 +316,43 @@ public void IsCompatibilitySwitchSet()
...
+                Assert.True(AppDomain.CurrentDomain.IsDefaultAppDomain());
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         public void IsFinalizingForUnload()
         {
-            Assert.False(AppDomain.CurrentDomain.IsFinalizingForUnload());
+            RemoteInvoke(() => {
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Why is the RemoteExecutor needed here?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 6]
```diff
@@ -268,28 +316,43 @@ public void IsCompatibilitySwitchSet()
...
+            }).Dispose();
         }
 
         [Fact]
         public void toString()
         {
             string actual = AppDomain.CurrentDomain.ToString();
+
+            // NetFx has additional line endings
+            if (PlatformDetection.IsFullFramework)
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Can you better state what is different in netfx pls. Line Endings like \r\n or \n wouldn't be removed by trim so I guess you mean whitespaces?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 7]
```diff
@@ -268,28 +316,43 @@ public void IsCompatibilitySwitchSet()
...
             string expected = "Name:" + AppDomain.CurrentDomain.FriendlyName + Environment.NewLine + "There are no context policies.";
             Assert.Equal(expected, actual);
         }
 
         [Fact]
         public void Unload()
         {
-            AssertExtensions.Throws<ArgumentNullException>("domain", () => { AppDomain.Unload(null);});
-            Assert.Throws<CannotUnloadAppDomainException>(() => { AppDomain.Unload(AppDomain.CurrentDomain); });
+            RemoteInvoke(() => {
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Why is the RemoteExecutor needed here? Looking through the netfx source code at least doesn't cause any side effects on the AppDomain.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 8]
```diff
@@ -313,15 +376,30 @@ public void LoadBytes()
         [Fact]
         public void ReflectionOnlyGetAssemblies()
         {
-            Assert.Equal(Array.Empty<Assembly>(), AppDomain.CurrentDomain.ReflectionOnlyGetAssemblies());
+            RemoteInvoke(() => {
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: The implementation is external so I can't or better I don't want to check the actual implementation but I guess this is just a read operation, why is the RemoteExecutor needed here?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 4 above)
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Actually since RETB derives from FCTB, you can use GetTestFilePath()

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 9]
```diff
@@ -313,15 +376,30 @@ public void LoadBytes()
...
         {
-            Assert.False(AppDomain.MonitoringIsEnabled);
-            AssertExtensions.Throws<ArgumentException>(null, () => {AppDomain.MonitoringIsEnabled = false;});
-            Assert.Throws<PlatformNotSupportedException>(() => {AppDomain.MonitoringIsEnabled = true;});
+            RemoteInvoke(() => {
+                Assert.False(AppDomain.MonitoringIsEnabled);
+                Assert.Throws<ArgumentException>(() => { AppDomain.MonitoringIsEnabled = false; });
+
+                if (PlatformDetection.IsFullFramework)
+                {
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Could you also check setting it to false here invokes an ArgumentException? https://referencesource.microsoft.com/#mscorlib/system/appdomain.cs,4770

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 10]
```diff
@@ -313,15 +376,30 @@ public void LoadBytes()
...
+            }).Dispose();
         }
 
         [Fact]
         public void MonitoringIsEnabled()
         {
-            Assert.False(AppDomain.MonitoringIsEnabled);
-            AssertExtensions.Throws<ArgumentException>(null, () => {AppDomain.MonitoringIsEnabled = false;});
-            Assert.Throws<PlatformNotSupportedException>(() => {AppDomain.MonitoringIsEnabled = true;});
+            RemoteInvoke(() => {
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Performance optimization for the infra tests would be to only invoke the RemoteExecutor for netfx.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 11]
```diff
@@ -339,20 +417,32 @@ public void MonitoringSurvivedProcessMemorySize()
         [Fact]
         public void MonitoringTotalAllocatedMemorySize()
         {
-            Assert.Throws<InvalidOperationException>(() => { var t = AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize; } );
+            RemoteInvoke(() => {
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Same as above:

> The implementation is external so I can't or better I don't want to check the actual implementation but I guess this is just a read operation, why is the RemoteExecutor needed here?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 12]
```diff
@@ -339,20 +417,32 @@ public void MonitoringSurvivedProcessMemorySize()
...
+                });
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         public void MonitoringTotalProcessorTime()
         {
-            Assert.Throws<InvalidOperationException>(() => { var t = AppDomain.CurrentDomain.MonitoringTotalProcessorTime; } );
+            Assert.Throws<InvalidOperationException>(() => {
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Same as above:

> The implementation is external so I can't or better I don't want to check the actual implementation but I guess this is just a read operation, why is the RemoteExecutor needed here?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 13]
```diff
@@ -339,20 +417,32 @@ public void MonitoringSurvivedProcessMemorySize()
...
+            } );
         }
 
 #pragma warning disable 618
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
         public void GetCurrentThreadId()
         {
-            Assert.True(AppDomain.GetCurrentThreadId() == Environment.CurrentManagedThreadId);
+            RemoteInvoke(() =>
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Why a RemoteExecutor here?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 14]
```diff
@@ -339,20 +417,32 @@ public void MonitoringSurvivedProcessMemorySize()
...
         {
-            Assert.Throws<InvalidOperationException>(() => { var t = AppDomain.CurrentDomain.MonitoringTotalProcessorTime; } );
+            Assert.Throws<InvalidOperationException>(() => {
+                var t = AppDomain.CurrentDomain.MonitoringTotalProcessorTime;
+            } );
         }
 
 #pragma warning disable 618
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Can you add a reason and/or issue here please

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 2 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: it couldn't copy or remove path. I didn't dig into that. I'm already improving current state by enabling those tests and since this is not that much related to UAP I decided to only do quick fixes or leave them as they were

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 15]
```diff
@@ -364,103 +454,131 @@ public void ShadowCopyFiles()
...
+                AppDomain.CurrentDomain.SetShadowCopyPath("test");
+                return SuccessExitCode;
+            }).Dispose();
         }
-#pragma warning restore 618
 
+#pragma warning restore 618
         [Fact]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Does not support Assembly.LoadFile")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Same as above. Can you add a reason and/or issue here please.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 16]
```diff
@@ -364,103 +454,131 @@ public void ShadowCopyFiles()
...
-            }
-            Assert.True(ctr > 0, "Assembly.LoadFile should cause file to be loaded again");
+                Assert.True(ctr > 0, "Assembly.LoadFile should cause file to be loaded again");
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Does not support Assembly.LoadFile")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Same as above. Can you add a reason and/or issue here please.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 17]
```diff
@@ -364,103 +454,131 @@ public void ShadowCopyFiles()
...
+                    AppDomain.CurrentDomain.AssemblyLoad -= handler;
+                }
+                Assert.True(AssemblyLoadFlag);
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         [ActiveIssue("https://github.com/dotnet/corefx/issues/18718", TargetFrameworkMonikers.Uap)] // Need to copy files out of execution directory'
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Access issues when copying assemblies")]
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Same as above, what kind of access issues?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 18]
```diff
@@ -482,6 +600,7 @@ public void AssemblyResolve()
 
         [Fact]
         [ActiveIssue("https://github.com/dotnet/corefx/issues/18718", TargetFrameworkMonikers.Uap)] // Need to copy files out of execution directory
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Access issues when copying assemblies")]
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Same as above, I will stop commenting on these now.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 19]
```diff
@@ -508,59 +627,68 @@ public void AssemblyResolve_RequestingAssembly()
...
         [Fact]       
         public void SetThreadPrincipal()
         {
-            Assert.Throws<ArgumentNullException>(() => {AppDomain.CurrentDomain.SetThreadPrincipal(null);});
-            var identity = new System.Security.Principal.GenericIdentity("NewUser");
-            var principal = new System.Security.Principal.GenericPrincipal(identity, null);
-            AppDomain.CurrentDomain.SetThreadPrincipal(principal);
+            RemoteInvoke(() => {
+                Assert.Throws<ArgumentNullException>(() => { AppDomain.CurrentDomain.SetThreadPrincipal(null); });
+                var identity = new System.Security.Principal.GenericIdentity("NewUser");
```
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: nit: GenericIdentity instead of var and GenericPrincipal instead of var

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 4 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: I did run uap tests before any additional changes and they were passing and did not run them after netfx changes since that requires clean build.

It is probably due to `[ActiveIssue(12716)]` I haven't spotted this. I have updated all tests which I noticed that modify any static state so that they use RemoteInvoke.

I'd leave this for the person fixing this issue since there might be more uap fixes we need here. I'll leave the comment in that issue

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 3 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: yes, I think there is also one more layer of abstraction somewhere since the AppDomainId seems to be 3 on desktop while it is 1 when you run it directly on netfx without any Xunits and other infra

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 5 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: It's checking static state which possibly may be changed by other test

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 6 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: @ViktorHofer Trim does remove `\r\n`:
`Console.WriteLine($"[[[{"\r\n".Trim()}]]]");` => `[[[]]]`

Netfx has exactly the same string as other platform except it has additional `\r\n`

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 5 above)
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Read operations shouldn't be required to be executed inside a RemoteExecutor instance as all write operations which change the state are *hopefully* executed inside a RemoteExecutor instance.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 7 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: It's trying to unload current appdomain. If this works on any platform because of the bug then lots of tests will fail, if you use RemoteInvoke only related will

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 6 above)
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Aaah thanks! Didn't know that :)

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 7 above)
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Good point!

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 10 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: If this works because of the bug, this will affect other test cases. Currently build time is considerably larger than all tests wrapped in RemoteInvoke. I wouldn't fight for few microseconds.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 19 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Unrelated change, I do not find it less readable despite the fact that I don't know this API.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 16 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: @ViktorHofer desktop seems to be doing that by design. It was disabled before without any reason and is still disabled without any reason provided. I do not consider this that much related to the purpose of this PR so I only went with quick fixes.

I think we should do large sweep in the end for all disabled issues without any reason and assign this work items to owners which can likely assess this much faster

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 13 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: `AppDomain.GetCurrentThreadId()` depends on the execution context

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 14 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: I left it unchanged. Was disabled before and after this change for no reason. I only added reasons where I could get quickly into a conclusion. I did not want to spend too much time on unrelated changes.

**Comment by krwq** [comment]
@dotnet-bot test OSX x64 Debug Build please (connection aborted)

**Comment by krwq** [comment]
FYI: locally passing UAP

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 13 above)
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: So is this failing outside a RemoteExecutor? I thought that the xunit execution context should still be the CurrentManagedThreadId?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 8 above)
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Like stated above in the other comment, tests which would affect others because they are changing states should be invoked inside an RemoteExeuctor instance. It doesn't make sense to "outsource" read operations. We haven't done this anywhere else.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 13 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: I'm not sure. There is almost no cost of running this remotely so not sure if this is worth investigating

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 5 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: I think it is cheaper to run sketchy tests remotely considering the execution time difference is measured in miliseconds or lesss

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 19 above)
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: coding guidelines say 'var' is fine when it's obvious what the type is eg when it's assigned 'new' like this.

Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Absolutely true. Missed that one.

**Comment by ViktorHofer** [comment]
I would like to see the few points regarding read operations inside a RemoteExecutor addressed and if somehow timely possible also the netfx skips. Just a suggestion, as we tend to move thing away and forget about them.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 15 above)
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: I'll leave this as is since I'm not disabling it and just leaving unchanged - there is some difference in behavior here which I don't want to investigate, since this has no reason it is really easy to grep. I'm hoping we will do some large sweep of all Skipped tests without a reason so that owners can help with understanding some differences

**Comment by krwq** [comment]
@ViktorHofer ok, I didn't investigate two remaining skipped netfx - better if someone knowing this area looks at it.

I reduced RemoteInvoke - I went slightly overboard with them (in some places I accidentally run RemoteInvoked inside RemoteInvoke)
I (hopefully) fixed access issues

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 20]
```diff
@@ -673,19 +677,18 @@ public void SetThreadPrincipal()
...
         {
             string destTestAssemblyPath = Path.Combine(Environment.CurrentDirectory, "AssemblyResolveTests", "AssemblyResolveTests.dll");
-            if (File.Exists("AssemblyResolveTests.dll"))
+            Console.WriteLine($"destinationPat={destTestAssemblyPath}");
+            if (!File.Exists(destTestAssemblyPath) && File.Exists("AssemblyResolveTests.dll"))
             {
                 Directory.CreateDirectory(Path.GetDirectoryName(destTestAssemblyPath));
-                File.Copy("AssemblyResolveTests.dll", destTestAssemblyPath, true);
-                File.Delete("AssemblyResolveTests.dll");
+                File.Copy("AssemblyResolveTests.dll", destTestAssemblyPath, false);
```
Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: I'm not sure why did we remove the file and forced overwrite - this works when there is one directory per tests but that's not what's happening on Desktop

**Comment by krwq** [comment]
@ViktorHofer, it seems that on the CI some of the assumptions in the test you wrote are wrong or no longer correct:

```
https://mc.dot.net/#/user/krwq/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/6b8793c68541020af66f5222d7083bfdcab6fdb8/workItem/System.Runtime.Extensions.Tests
```

Seems like those 3 folders are not available - is that a bug?
```
GetFolderPath_Windows(folder: AdminTools, existsInAppContainer: True) 
GetFolderPath_Windows(folder: Programs, existsInAppContainer: True)
GetFolderPath_Windows(folder: Startup, existsInAppContainer: True) 
```

I'll check what is the correct list

**Comment by danmosemsft** [comment]
@krwq don't worry too much about what's available - just make the test pass. We are waiting for guidance from Windows on what/how we shuld call to get better data: https://github.com/dotnet/corefx/issues/20782. We'll need to update the test then. We probably won't ship with what we have here.

**Comment by krwq** [comment]
talked with @JeremyKuhne offline - seems like this is still being discussed and has changed. I'll update according to newest or close with active issue while discussion is pending.

**Comment by krwq** [comment]
@dotnet-bot test this please

**Comment by krwq** [comment]
@dotnet-bot test OSX x64 Debug Build please

**Comment by mmitche** [comment]
@dotnet-bot help

**Comment by mmitche** [comment]
@dotnet-bot help (testing)

**Comment by krwq** [comment]
@dotnet-bot test Linux x64 Release Build please

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 1 above)
Comment (jkotas) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Number of these methods are not-yet-implemented for UapAot. The tests should be disabled against issues to get them implemented.

Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Thanks @jkotas, will check this

Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: @jkotas uapaot does not work for me locally, I tried on the current state and after rebasing to newest and I keep getting this failure:

```
System.Private.CoreLib : Gatekeeper error GK0028 : GK0028 Assembly 'System.Private.CoreLib' is a mixed mode assembly containing managed and native code.  Only pure managed assemblies are compatible with .NET Native.
```

Comment (krwq) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Here is how I build:
```
set OptionalToolSource=https://devdiv.pkgs.visualstudio.com/_packaging/dotnet-core-internal-tooling/nuget/v3/index.json
set OptionalToolSourceUser=<myuser>
set OptionalToolSourcePassword=<PAT>
sync
build -framework:uapaot
```

and then building test with:
```
msbuild /t:RebuildAndTest /p:TargetGroup=uapaot System.Runtime.Extensions.Tests.csproj
```

Comment (jkotas) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: @danmosemsft What is the best workaround for this crash?

**Comment by krwq** [comment]
Thanks to @joperezr's help I have disabled failures on uapaot. Issue:
https://github.com/dotnet/corefx/issues/21680

**Comment by danmosemsft** [comment]
Windows failure is https://github.com/dotnet/corefx/issues/20457

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: `ystem.Private.CoreLib : Gatekeeper error GK0028 : GK0028 Assembly 'System.Private.CoreLib' is a mixed mode assembly containing managed and native code.  Only pure managed assemblies are compatible with .NET Native.`  is caused by https://github.com/dotnet/buildtools/issues/1587

it happens when swtiching between running netcoreapp and aot tests on the same netstandard library. The workaround is to blow the folder away and rebuild.
